hail/hail/src/is/hail/expr/ir/lowering/LowerBlockMatrixIR.scala

```diff
@@ -1145,6 +1145,64 @@ object LowerBlockMatrixIR {
     bmir match {
       case BlockMatrixRead(reader) => reader.lower(ctx, ib)
 
+      case ValueToBlockMatrix(child, _, _)
+          if !child.typ.isInstanceOf[TArray] && !child.typ.isInstanceOf[TNDArray] =>
+        val element = LowerToCDA.lower(child, typesToLower, ctx, analyses)
+        val contextsIR = MakeArray(MakeStruct(FastSeq()))
+
+        def blockIR(ctxRef: Ref): IR = MakeNDArray(
+          MakeArray(element),
+          MakeTuple(FastSeq((0, I64(1)), (1, I64(1)))),
+          False(),
+          ErrorIDs.NO_ERROR,
+        )
+
+        BlockMatrixStage2(
+          FastSeq(),
+          bmir.typ,
+          BMSContexts(bmir.typ, contextsIR, ib),
+          blockIR,
+        )
+
+      case x @ ValueToBlockMatrix(child, _, _) =>
+        val lowered = LowerToCDA.lower(child, typesToLower, ctx, analyses)
+        val nd = ib.memoize(child.typ match {
+          case _: TArray => MakeNDArray(
+              lowered,
+              MakeTuple.ordered(FastSeq(I64(x.typ.nRows), I64(x.typ.nCols))),
+              True(),
+              ErrorIDs.NO_ERROR,
+            )
+          case _: TNDArray => lowered
+        })
+        val contexts = BMSContexts.tabulate(x.typ, ib) { (rowBlockIdx, colBlockIdx) =>
+          val rowStartIdx = rowBlockIdx.toL * I64(x.typ.blockSize)
+          val colStartIdx = colBlockIdx.toL * I64(x.typ.blockSize)
```

Comment:
Should these be let-bound to avoid reevaluation?

Reply:
yes? https://github.com/hail-is/hail/pull/15086/commits/23ad6e9212bbe73cf6924d7415eb33c5f350571f

Reply:
Ahhh you used `bindIRs` instead of the ir builder, `ib`! Thanks